db/sqlglue.c

```diff
@@ -4831,7 +4831,7 @@ int sqlite3BtreeBeginTrans(Vdbe *vdbe, Btree *pBt, int wrflag)
 #endif
 
     /* already have a transaction, keep using it until it commits/aborts */
-    if (clnt->intrans || clnt->no_transaction ||
+    if (clnt->intrans || vdbe == NULL ||
```

Comment:
Does this handle cases of `reload_analyze` under snapshot isolation? Would clnt->intrans be set to 1 later in this function? (I think [Line 4901](https://github.com/bloomberg/comdb2/pull/303/files#diff-c3ebbfffbf6cf6842b5727af7e3a5e99R4901) only checks for mode <= RECOM,)